homework/index.js

```diff
@@ -30,14 +30,105 @@
+    createAndAppend('td', tr, { text: `${labelText}`, class: str });
+    createAndAppend('td', tr, { text: value });
+    return tr;
+  }
+
+  function renderRepoInformation(repositories, container) {
+    container.innerHTML = ' ';
+    const table = createAndAppend('table', container, { class: 'infoTable' });
+    const tbody = createAndAppend('tbody', table);
+    const firstRow = addRow(tbody, 'Repository: ', '', 'explanations');
+    createAndAppend('a', firstRow.lastChild, {
+      href: `${repositories.html_url}`,
+      target: 'blank',
+      text: `${repositories.name}`,
+    });
+    addRow(tbody, 'Description:', repositories.description, 'explanations');
+    addRow(tbody, 'Fork:', repositories.forks, 'explanations');
+    addRow(tbody, 'Updated:', `${repositories.updated_at}`.substring(0, 10), 'explanations');
+  }
+
+  function renderContributorsInformation(repositories, container) {
+    const contributors = repositories.contributors_url;
+    const url = contributors;
+    fetchJSON(url, (error, contributorDetails) => {
+      if (error) {
+        renderError(error);
+        return;
+      }
+
+      container.innerHTML = ' ';
+      const div = document.getElementById('cont-ulist');
+      createAndAppend('h4', div, { text: 'Contributors' });
+      const contrList = createAndAppend('ul', div, { class: 'contr-list' });
+      contributorDetails.forEach(contributor => {
+        const li = createAndAppend('li', contrList);
+        const a = createAndAppend('a', li, { href: contributor.html_url, target: 'blank' });
+        createAndAppend('img', a, {
+          src: contributor.avatar_url,
+          class: 'images',
+        });
+        createAndAppend('span', a, {
+          text: contributor.login,
+        });
+        createAndAppend('span', a, {
+          text: contributor.contributions,
+          class: 'contributions',
+        });
+      });
+    });
+  }
+
   function main(url) {
+    const root = document.getElementById('root');
+    const menuSection = createAndAppend('section', root, { id: 'menu-section' });
+    createAndAppend('p', menuSection, { text: 'HYF Repositories', id: 'hyf-text' });
+    const select = createAndAppend('select', menuSection, { id: 'select-button' });
+    const bodyDiv = createAndAppend('div', root, { id: 'body-div' });
+    const section = createAndAppend('section', bodyDiv, { class: 'repo-info-list' });
+    const contrDiv = createAndAppend('div', bodyDiv, {
+      class: 'contributors-div',
+      id: 'cont-ulist',
+    });
+
     fetchJSON(url, (err, repositories) => {
-      const root = document.getElementById('root');
+      repositories.sort((a, b) => a.name.localeCompare(b.name));
       if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+        renderError(err);
         return;
       }
-      createAndAppend('pre', root, { text: JSON.stringify(repositories, null, 2) });
+
+      function showFirstRepo() {
```

Comment:
This immediately invoked function looks a lot like what I have seen in the homework from @gulyener. As is it stands, this function does not add much value over simply executing lines 120-121 directly, e.g.:

```js
fetchJSON(url, (err, repositories) => {
  repositories.sort((a, b) => a.name.localeCompare(b.name));
  if (err) {
    renderError(err);
    return;
  }

  renderRepoInformation(repositories[0], section);
  renderContributorsInformation(repositories[0], contrDiv);

  createOptions(repositories, select);
  select.addEventListener('change', () => {
```

If you want to use a function, make it reusable like this:

```js
function showRepo(repo, repoContainer, contributorContainer) {
  renderRepoInformation(repo, repoContainer);
  renderContributorsInformation(repo, contributorContainer);
}

function main(url) {
   ...
  fetchJSON(url, (err, repositories) => {
    if (err) {
      renderError(err);
      return;
    }
    repositories.sort((a, b) => a.name.localeCompare(b.name));
    createOptions(repositories, select);

    showRepo(repositories[0], section, contrDiv);

    select.addEventListener('change', () => {
      const repository = repositories[select.value];
      showRepo(repository, section, contrDiv);
    });
  });
}
```

Reply:
So true, I don't know why I felt the need of using a non-reusable function. I've just realized that I don't need a specific function for showing the first repository, so I am just going to call my two functions instead of calling them inside of a new function. Is it alright?